benchmark/run_benchmarks_jump.jl

```diff
@@ -0,0 +1,314 @@
+using MathOptInterface
+using ParametricOptInterface
+using BenchmarkTools
+using JuMP
+const MOI = MathOptInterface
+const POI = ParametricOptInterface
+import Pkg
+
+function moi_add_variables(N::Int)
+    model = direct_model(MOIU.MockOptimizer(MOI.Utilities.Model{Float64}()))
+    @variable(model, x[i=1:N])
+    return nothing
+end
+
+function poi_add_variables(N::Int)
+    model = direct_model(POI.Optimizer(MOIU.MockOptimizer(MOI.Utilities.Model{Float64}())))
+    @variable(model, x[i=1:N])
+    return nothing
+end
+
+function poi_add_parameters(N::Int)
+    model = direct_model(POI.Optimizer(MOIU.MockOptimizer(MOI.Utilities.Model{Float64}())))
+    @variable(model, x[i=1:N] in POI.Parameter(1))
+    return nothing
+end
+
+function poi_add_parameters_and_variables(N::Int)
+    model = direct_model(POI.Optimizer(MOIU.MockOptimizer(MOI.Utilities.Model{Float64}())))
+    @variable(model, x[i=1:N] in POI.Parameter(1))
+    return nothing
+end
+
+function poi_add_parameters_and_variables_alternating(N::Int)
+    model = direct_model(POI.Optimizer(MOIU.MockOptimizer(MOI.Utilities.Model{Float64}())))
+    for i in 1:Int(N/2)
+        @variable(model)
+        @variable(model, set = POI.Parameter(1))
+    end
+    return nothing
+end
+
+function moi_add_saf_ctr(N::Int, M::Int)
+    model = direct_model(MOIU.MockOptimizer(MOI.Utilities.Model{Float64}()))
+    @variable(model, x[i=1:N])
+    @constraint(model, cons[i=1:M], sum(x) >= 1)
+    return nothing
+end
+
+function poi_add_saf_ctr(N::Int, M::Int)
+    model = direct_model(POI.Optimizer(MOIU.MockOptimizer(MOI.Utilities.Model{Float64}())))
+    @variable(model, x[i=1:N])
+    @constraint(model, cons[i=1:M], sum(x) >= 1)
+    return nothing
+end
+
+function poi_add_saf_variables_and_parameters_ctr(N::Int, M::Int)
+    model = direct_model(POI.Optimizer(MOIU.MockOptimizer(MOI.Utilities.Model{Float64}())))
+    @variable(model, x[i=1:Int(N/2)])
+    @variable(model, p[i=1:Int(N/2)] in POI.Parameter.(0))
+    @constraint(model, con[i=1:M], sum(x) + sum(p) >= 1)
+    return nothing
+end
+
+function poi_add_saf_variables_and_parameters_ctr_parameter_update(N::Int, M::Int)
+    model = direct_model(POI.Optimizer(MOIU.MockOptimizer(MOI.Utilities.Model{Float64}())))
+    @variable(model, x[i=1:Int(N/2)])
+    @variable(model, p[i=1:Int(N/2)] in POI.Parameter.(0))
+    @constraint(model, con[i=1:M], sum(x) + sum(p) >= 1)
+    MOI.set.(model, POI.ParameterValue(), p, 0.5)
+    POI.update_parameters!(model.moi_backend)
+    return nothing
+end
+
+function moi_add_sqf_variables_ctr(N::Int, M::Int)
+    model = direct_model(MOIU.MockOptimizer(MOI.Utilities.Model{Float64}()))
+    @variable(model, x[i=1:N])
+    @constraint(model, con[i=1:M], sum(x.^2) >= 1)
```

Comment:
Could you try to see if `dot` is better than this sum

Reply:
I would not be surprised if `sum(x[i]^2 for i in eachindex(x))` was the fastest.

Reply:
```
SQF constraint with variables on a POI.Optimizer.
  1.098 s (13070741 allocations: 1.38 GiB)
SQF dot constraint with variables on a POI.Optimizer.
  384.832 ms (70541 allocations: 323.04 MiB)
SQF foreach constraint with variables on a POI.Optimizer.
  950.255 ms (13066341 allocations: 1.24 GiB)
```
`dot` is the winner